lib/pump.c

```diff
@@ -0,0 +1,135 @@
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+
+#define NUM_SENSORS 5
+
+#define PUMP_THRESHOLD_VERYHIGH  20
+#define PUMP_THRESHOLD_HIGH  15
+#define PUMP_THRESHOLD_LOW  10
+#define PUMP_THRESHOLD_VERYLOW  5
+
+
+ struct DataStruct
+  {
+      int id;
+      int data;
+  } pump_data;
+int table [5][2];
+bool pumpeisON = false;
+void resetTable( int table[][2]){
+
+	for (int i=0;i<NUM_SENSORS;i++){
+		table[i][0]=0;
+		table[i][1]=0;
+	}
+}
+
+void printTable( int table[][2]){
+
+	for(int i=0;i<NUM_SENSORS;i++){
+		printf("id: %d value: %d \n",table[i][0],table[i][1]);
+	}
+}
+
+
+bool mainPump(struct DataStruct pump_data, bool pumpeisON){
+
+    int openPumpe = 0;
+    int closePumpe = 0;
+    int sumHum = 0;
+    int avgHum = 0;
+    if (pump_data.data  < PUMP_THRESHOLD_VERYLOW || pump_data.data > PUMP_THRESHOLD_VERYHIGH){
+
+        if(pump_data.data < PUMP_THRESHOLD_VERYLOW && !pumpeisON){
+		printf("OpenPumpe \n");
+		resetTable(table);
+		pumpeisON = true;
+            //send(OpenPumpe);
+        }
+
+        if(pump_data.data  > PUMP_THRESHOLD_VERYHIGH && pumpeisON){
+	      printf("ClosePumpe");
+	      resetTable(table);
+             //send(ClosePumpe);
+        }
+
+
+    }
+
+    else{
+        bool repeatedData = false;
+        for(int i=0;i<NUM_SENSORS-1;i++){
+
+            if(table[i][0]==pump_data.id){
+                repeatedData = true;
+		table[i][1] = pump_data.data;
+		printf("TableUpdated \n");
+		printTable(table);
+            }
+        }
+        if(!repeatedData){
+	    int aux=0;
+	    while(table[aux][0] != 0){
+		aux++;
+	    }
+            table[aux][0] = pump_data.id;
```

Comment:
you could move all of this to the for loop above, then you can just `table[i][0] = ` instead of that extra while loop that calculates `aux`